src/include/homestore/chunk_selector.h

```diff
@@ -0,0 +1,30 @@
+/*********************************************************************************
+ * Modifications Copyright 2017-2023 eBay Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <homestore/vchunk.h>
+
+namespace homestore {
+class Chunk;
+class ChunkSelector {
+public:
+    ChunkSelector() = default;
+    virtual void add_chunk(VChunk chunk) = 0;
+    virtual void foreach_chunks(std::function< void(VChunk&) >&& cb) = 0;
+    virtual Chunk* select_chunk(blk_count_t nblks, const blk_alloc_hints& hints) = 0;
```

Comment:
Is there any challenge or reason that we need to expose two different types VChunk/Chunk* to consumer with add/select? 
Can we either return VChunk or Chunk for both add/select apis?

Reply:
customer code can use
```
ChunkSelector::add_chunk(chunk *)
chunk *p_chunk = ChunkSelector:select()
```

Reply:
for now, add_chunk(cshared< Chunk >& chunk) is actually used in homestore. if we pass raw pointer directly, this might case a pointer leaker, since a raw pointer is manager by different shared_ptr.

if we want to use `add_chunk(chunk *)` , we need to change the use style in homestore

Reply:
@yamingk i change it to return VChunk in both add/select apis to keep the style consistent, and do the transition inside homestore